File: Software Development Module/Week9/ResultsApp.java
/*
ResultsApp.java
Billy Cussen
18/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class ResultsApp {
    public static void main(String[]args){

        //Variables
        int[][] examResult;
        int numberOfStudents = 0;
        int numberOfExams = 0;
        String result;
        boolean inputCheckResults;

        //Objects
        Scanner keyboard;
        Results myResults;

        //Input
        do{
            inputCheckResults = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter the number of Students to Grade: ");
                numberOfStudents = keyboard.nextInt();

                if(numberOfStudents <= 0){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("You need to enter an Integer (of 1 or greater) to Register how many Student you wish to Grade");
                inputCheckResults = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheckResults);

        do{
            inputCheckResults = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter the number of Exams to Grade: ");
                numberOfExams = keyboard.nextInt();

                if(numberOfExams <= 0){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("You need to enter an Integer (of 1 or greater) to Register how many Exams you wish to Grade");
                inputCheckResults = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheckResults);

        examResult = new int[numberOfStudents][numberOfExams];

        for(int i = 0; i < examResult.length; i++){

            for(int j = 0; j < examResult[0].length; j++){

                do{
                    inputCheckResults = true;
                    keyboard = new Scanner(System.in);
        
                    try{
                        System.out.print("Please enter the Grade for Student "+(i+1)+", Exam "+(j+1)+": ");
                        examResult[i][j] = keyboard.nextInt();

                        if(examResult[i][j] < 0 || examResult[i][j] > 100){
                            throw new InputMismatchException();
                        }
                    }catch(InputMismatchException e){
                        System.out.println("You need to enter an Integer (of 0 or greater and less than 101) to Register these exam results");
                        inputCheckResults = false;      
                    }finally{
                        keyboard = null;
                    }
        
                }while(!inputCheckResults);

            }

        }

        myResults = new Results(examResult);
        
        //Process & Output
        result = myResults.computeAndGetResults();
        myResults = null;
        System.out.println(result);

    }

}

File: Python Course/Software Development Module - in Python/Week9/ResultsApp.py
"""
ResultsApp.py
Billy Cussen
01/10/2020
"""

#Objects
import gc
import Results
from Results import myResults

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = True
    #Input - Get Number of Students
    while inputCheck:
        try:
            students = int(input("Please enter the number of Students being checked (10 Max): "))
        except:
            print("You need to enter an Integer between 1 and 10")
        else:
            if students < 1 or students > 10:
                print ("You need to pick between 1 and 10 students")
            else:
                inputCheck = False
    
    #Input - Get Exams to Grade
    while inputCheck == False:
        try:
            exams = int(input("Please enter the number of exams to Grade (5 max): "))
        except:
            print("You need to enter an Integer between 1 and 5")
        else:
            if exams < 1 or exams > 5:
                print("You need to pick between 1 and 5 exams")
            else:
                inputCheck = True

    #Input - Get Results of each Exam for each Student (Using a 2D Array)
    rows, columns = (students, exams)
    examResultsAll = []
    for i in range(columns):
        examResults = []
        for j in range(rows):
            inputCheck = False
            while inputCheck == False:
                try:
                    resultTemp = int(input("Please enter results for Exam "+str(i+1)+" for Student "+str(j+1)+": "))
                except:
                    print("You need to enter an Integer for their Grade (e.g. 45, 92)")
                else:
                    if resultTemp < 0 or resultTemp > 100:
                        print("The result entered needs to be between 0 and 100")
                    else:
                        inputCheck = True
                        examResults.append(resultTemp)
        
        examResultsAll.append(examResults)
    
    myResultsApp = myResults(examResultsAll)
    myResultsApp.computeAndGetResults()
    print(myResultsApp.result)
    myResultsApp = None
    gc.collect()

    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"Y\" or \"N\" if you would like to Repeat or Exit: "))
        repeat = repeat.lower()

        if repeat != "y" and repeat != "n":
            print("You need to enter \"Y\" or \"N\"")

print("Thanks for using the App!")